src/actions/userProfile.js

```diff
@@ -73,6 +76,31 @@ export const updateUserProfileProperty = (userProfile, key, value) => {
   };
 };
 
+export const getProjectsByUsersName = searchName => {
+  const url = ENDPOINTS.GET_PROJECT_BY_PERSON(searchName);
+  return async dispatch => {
+    try {
+      const res = await axios.get(url);
+      await dispatch(getProjectsByPersonActionCreator(res.data));
+      return res.data.allProjects;
+    } catch (error) {
+      await dispatch(userNotFoundError(error.message));
```

Comment:
await may not be used directly with dispatch because dispatch doesn't return a Promise. The thunk function itself returns a Promise due to the asynchronous operation, but the dispatch call within it executes synchronously.

Reply:
Thank you for pointing that out!